File: benchmark/results/python-multilang/run-004/b-baseline/output/metaengine_demo/billing/services/invoice_service.py
"""InvoiceService for the billing domain."""
from typing import Optional

from metaengine_demo.billing.aggregates.invoice import Invoice


class InvoiceService:
    """Service exposing invoice operations."""

    def create(self, input: Optional[Invoice]) -> Invoice:
        """Create a new Invoice."""
        raise NotImplementedError("not implemented")

    def findById(self, id: str) -> Optional[Invoice]:
        """Find an Invoice by id."""
        raise NotImplementedError("not implemented")

    def list(self, limit: float) -> list[Invoice]:
        """List Invoices up to limit."""
        raise NotImplementedError("not implemented")

    def delete(self, id: str) -> None:
        """Delete an Invoice by id."""
        raise NotImplementedError("not implemented")

File: benchmark/results/java-multilang/run-001/b-baseline/output/com/metaengine/demo/billing/services/InvoiceService.java
package com.metaengine.demo.billing.services;

import java.util.List;

import com.metaengine.demo.billing.aggregates.Invoice;

/** InvoiceService for the billing domain. */
public class InvoiceService {

    /** Creates a new Invoice from the partial input. */
    public Invoice create(Invoice input) {
        throw new UnsupportedOperationException("not implemented");
    }

    /** Finds an Invoice by id. May return null. */
    public Invoice findById(String id) {
        throw new UnsupportedOperationException("not implemented");
    }

    /** Lists Invoices up to the given limit. */
    public List<Invoice> list(int limit) {
        throw new UnsupportedOperationException("not implemented");
    }

    /** Deletes the Invoice with the given id. */
    public void delete(String id) {
        throw new UnsupportedOperationException("not implemented");
    }
}
